core/templates/dev/head/pages/exploration_editor/editor_tab/StateHints.js

```diff
@@ -57,6 +57,11 @@ oppia.controller('StateHints', [
       var hintAsPlainText = (
         hint.hintContent.getHtml() ?
           $filter('convertToPlainText')(hint.hintContent.getHtml()) : '');
+      if (hintAsPlainText === '') {
+        hintAsPlainText = hint.hintContent.getHtml().split(
```

Comment:
This seems ... wrong? What if the hint has more than one rich-text component, plus some other text, etc.?

I suggest taking a look at my comments on #4453, including the reference to [this approach](https://github.com/oppia/oppia/blob/develop/core/templates/dev/head/expressions/ExpressionInterpolationService.js#L44) (using string.replace()). Perhaps the common functionality could go in a separate filter so that @Nix07 can use it too.

Reply:
Done